drafting table finishing touches

EvadLhorg opened the pull request.

mmeldo committed:
Add promptId to AbilityTimer expire func (#148)

ystros committed:
Add support for game links and private games (#149)

* Remove modals from new / pending games

* Allow games to be joined via game link

Also adds support for private games

EvadLhorg committed:
update throneteki-deck-helper to 2.5.1 (#151)

EvadLhorg committed:
display plots in drafting hand with horizontal orientation

EvadLhorg committed:
update throneteki-deck-helper

EvadLhorg committed:
display faction icons in the drafting deck builder
fix some style usages with the correct style names
always use the event restricted list in the draft deck editor instead of the one chosen in the deck list view

ystros commented on Components/Decks/DraftDeckEditor.jsx:
```diff
@@ -80,7 +80,7 @@ class DraftDeckEditor extends React.Component {
         return remainingCards;
     }
 
-    getDeckFromState(restrictedList) {
```
There's still an existing call to this method passing restrictedList

EvadLhorg replied:
fixed

ystros commented on Components/Decks/DraftDeckEditor.jsx:
```diff
@@ -359,7 +360,7 @@ class DraftDeckEditor extends React.Component {
                 </div>
                 <div className='row'>
-                    <div className='col-sm-12' style={ { 'text-align': 'center', 'margin-bottom': '1em' } }>
+                    <div className='col-sm-12' style={ { 'textAlign': 'center', 'marginBottom': '1em' } }>
```
I think the quotes are no longer needed if they're camel-cased.

EvadLhorg replied:
also fixed

ystros commented on Components/Decks/DraftDeckEditor.jsx:
```diff
@@ -97,7 +97,8 @@ class DraftDeckEditor extends React.Component {
             deck.status = {};
         } else {
-            const selectedRestrictedList = restrictedList || this.props.currentRestrictedList;
+            const draftEventRestrictedList = this.props.restrictedList.filter(rl => rl._id === this.props.deck.eventId)[0];
```
Thinking a bit, should we even be validating against a restricted list for draft decks? I think in the current admin for a draft event, there's no way to even select a RL. The only reason we'd need most of the code here is if we ever wanted to support a draft event that did have a RL.

If not, we could just send an empty array for the restrictedLists property (this is what I've done locally for testing purposes)

EvadLhorg replied:
I agree that at the moment a restricted list would not be necessary. Using the code i committed (which looks for the RL of the event/draft event) leads to the same behaviour that you tested with locally. So i would say it wouldn´t hurt to use the event RL (which is always empty at the moment) so we have to change less stuff when we enable custom RL for draft events.
what do you think?

ystros reviewed (commented):
Thanks! Looks mostly good, though there are conflicts in addition to my comments below.

EvadLhorg committed:
remove restrictedList parameter from triggerDeckUpdated and remove unnecessary quotes in style property

EvadLhorg committed:
Merge branch 'features/drafting-table' of https://github.com/ystros/throneteki-client into features/drafting-table-2

EvadLhorg committed:
remove dispatch call for actions that are already calling dispatch implicitly due to being boundActions (#161)

dependabot[bot] committed:
Bump jszip from 3.1.5 to 3.7.1 (#152)

Bumps [jszip](https://github.com/Stuk/jszip) from 3.1.5 to 3.7.1.
- [Release notes](https://github.com/Stuk/jszip/releases)
- [Changelog](https://github.com/Stuk/jszip/blob/master/CHANGES.md)
- [Commits](https://github.com/Stuk/jszip/compare/v3.1.5...v3.7.1)

---
updated-dependencies:
- dependency-name: jszip
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ua-parser-js from 0.7.18 to 0.7.31 (#154)

Bumps [ua-parser-js](https://github.com/faisalman/ua-parser-js) from 0.7.18 to 0.7.31.
- [Release notes](https://github.com/faisalman/ua-parser-js/releases)
- [Commits](https://github.com/faisalman/ua-parser-js/compare/0.7.18...0.7.31)

---
updated-dependencies:
- dependency-name: ua-parser-js
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump pathval from 1.1.0 to 1.1.1 (#156)

Bumps [pathval](https://github.com/chaijs/pathval) from 1.1.0 to 1.1.1.
- [Release notes](https://github.com/chaijs/pathval/releases)
- [Changelog](https://github.com/chaijs/pathval/blob/master/CHANGELOG.md)
- [Commits](https://github.com/chaijs/pathval/compare/v1.1.0...v1.1.1)

---
updated-dependencies:
- dependency-name: pathval
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump handlebars from 4.1.2 to 4.7.7 (#157)

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.1.2 to 4.7.7.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/handlebars-lang/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.1.2...v4.7.7)

---
updated-dependencies:
- dependency-name: handlebars
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

EvadLhorg committed:
Add locked deck feature for events (#166)

* WIP selecting an event for a deck and prevent event decks from being changed

* check for deck.locked rather than eventId to determine if a deck can be editted

* display the event name in the decklist so a user can easily see which deck is used for which event

* fix proptype

* pass events to selectDeckModal so that DeckRow works in every instance

* add a lockDecks property to an event

* change labels on the deckrow and event editor
use string interpolation in deckrow to add the event name to a deck name

* prevent undefined deckEvent usage

* filter the events to lockDecks events only
only show the event dropdown when there are events to be selected

EvadLhorg committed:
update throneteki deckhelper to 2.6.0 (#167)

cryogen committed:
Fix errors/warnings

EvadLhorg committed:
fix the deck selection for games that use an event but the event is not locking the decks (#168)

Patane97 committed:
Merging AHaH-related updates (#181)

* Add jail token for Mord

* enable "show Hand" by default for new games

* add a review button that links to the card review form to the game won prompt

* Add initiative to the player stat bar

* Adding highlight for cards contributing STR

* Removing unused Jail token

* Removing remaining jail code & reverting showHands

* Removing review button on game end

* Bump deck helper for AHaH validations

Co-authored-by: Brian Upton <ystros@gmail.com>
Co-authored-by: EvadLhorg <lukas.a-v@web.de>

EvadLhorg committed:
extend the event editor so that an admin can define pods for a custom restricted list. pods must be in json format, the same that is used in the restricted-list.json file of the throneteki-json-data repo (#185)

EvadLhorg committed:
display a warning icon on the navbar next to the spectators popup when a spectator joins/leaves the game (#187)

Patane97 committed:
Implementing changes as part of TSoW & engine updates (#188)

* Initial commit for TSoW

* Adding highlight for assaulted cards

* Updating review prompt to process dynamic form id

* Fixing integer arg bug for googleFormMatcher

* Fix AbilityTargeting to show generic shadow data

EvadLhorg committed:
add an option to games and events to set the delay seconds when playing with chess clocks (#189)

* add an option to games and events to set the delay seconds when playing with chess clocks
fix the chessclock display so that it doesn´t "shake" the screen when counting down on some screen resolutions/zoom stages

* Tweaking/refining the visuals of the delay timer

* fix linter error, multiline jsx was missing ( )

* stop the clock from counting down when the time has expired

* when activating a time limit option in the event editor, deactivate the other one

* Updating node-sass to 4.14.1

* Updating package-lock to v2 & node-sass to 6.0.1

---------

Co-authored-by: Stephen Patane <patane97@gmail.com>

EvadLhorg committed:
Merge branch 'master' of https://github.com/throneteki/throneteki-client into features/drafting-table-2

EvadLhorg committed:
fix the draft prompt area so it is always displayed completely instead of with a scrollbar

EvadLhorg committed:
check for different properties to determine if a deck can be deleted or edited
